src/js/tabs/motors.js

```diff
@@ -703,34 +936,89 @@ TABS.motors.initialize = function (callback) {
-                $(`.servo-${i} .indicator`, servos_wrapper).css({
-                    'margin-top' : `${marginTop}px`,
-                    'height' : `${height}px`,
-                    'background-color' : `rgba(255,187,0,1${color})`,
-                });
-            }
             //keep the following here so at least we get a visual cue of our motor setup
             update_arm_status();
 
             if (previousArmState != self.armed) {
                 console.log('arm state change detected');
 
-                $('#motorsEnableTestMode').change();
+                motorsEnableTestModeElement.change();
             }
         }
 
+        $('a.save').on('click', function() {
+            // gather data that doesn't have automatic change event bound
+            FC.MOTOR_CONFIG.minthrottle = parseInt($('input[name="minthrottle"]').val());
+            FC.MOTOR_CONFIG.maxthrottle = parseInt($('input[name="maxthrottle"]').val());
+            FC.MOTOR_CONFIG.mincommand = parseInt($('input[name="mincommand"]').val());
+
+            if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_42)) {
+                FC.MOTOR_CONFIG.motor_poles = parseInt($('input[name="motorPoles"]').val());
+            }
+
+            FC.MOTOR_3D_CONFIG.deadband3d_low = parseInt($('input[name="3ddeadbandlow"]').val());
+            FC.MOTOR_3D_CONFIG.deadband3d_high = parseInt($('input[name="3ddeadbandhigh"]').val());
+            FC.MOTOR_3D_CONFIG.neutral = parseInt($('input[name="3dneutral"]').val());
+
+            FC.PID_ADVANCED_CONFIG.fast_pwm_protocol = parseInt(escProtocolElement.val() - 1);
+            FC.PID_ADVANCED_CONFIG.use_unsyncedPwm = unsyncedPWMSwitchElement.is(':checked') ? 1 : 0;
+            FC.PID_ADVANCED_CONFIG.motor_pwm_rate = parseInt($('input[name="unsyncedpwmfreq"]').val());
+            FC.PID_ADVANCED_CONFIG.digitalIdlePercent = parseFloat($('input[name="digitalIdlePercent"]').val());
+
+            if (semver.gte(FC.CONFIG.apiVersion, "1.25.0") && semver.lt(FC.CONFIG.apiVersion, API_VERSION_1_41)) {
+                FC.PID_ADVANCED_CONFIG.gyroUse32kHz = $('input[id="gyroUse32kHz"]').is(':checked') ? 1 : 0;
+            }
+
+            analytics.sendChangeEvents(analytics.EVENT_CATEGORIES.FLIGHT_CONTROLLER, self.analyticsChanges);
+            self.analyticsChanges = {};
+
+            const mspCommands = [
+                MSPCodes.MSP_SET_ARMING_CONFIG,
+                MSPCodes.MSP_SET_MOTOR_3D_CONFIG,
+                MSPCodes.MSP_SET_MOTOR_CONFIG,
+                MSPCodes.MSP_SET_ADVANCED_CONFIG,
+                MSPCodes.MSP_SET_FEATURE_CONFIG,
+                MSPCodes.MSP_SET_MIXER_CONFIG,
+            ];
+
+            if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_31)) {
+                mspCommands.push(MSPCodes.MSP_SET_FILTER_CONFIG);
+            }
+
+            let commands = mspCommands.length - 1;
+
+            function saveSettings() {
+                MSP.send_message(
```

Comment:
I am sorry but this is very hard to understand - I think using `MSP.promise` will make this a lot easier to understand.
Is my guess right that the list in `mspCommands` is executed backwards? Why?
Also, n.b. that order matters for MSP commands setting configuration, as some of them have an overlap.

Reply:
Changed to promise